src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,53 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 public class SalaryInfo {
+    private static final String DATE_PATTERN = "dd.MM.yyy";
+    private static final short LINE_DATE = 0;
+    private static final short LINE_NAME = 1;
+    private static final short LINE_HOURS = 2;
+    private static final short LINE_SALARY = 3;
+    private final SimpleDateFormat simpleDateFormat = new SimpleDateFormat(DATE_PATTERN);
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder report = new StringBuilder("Report for period ")
+                .append(dateFrom).append(" - ").append(dateTo).append(System.lineSeparator());
+        for (String name : names) {
+            report.append(name).append(" - ")
+                    .append(getSalary(name, data, dateFrom, dateTo)).append(System.lineSeparator());
+        }
+        return report.toString().trim();
+    }
+
+    private int getSalary(String name, String[] data, String fromDate, String toDate) {
```

Comment:
`getSalary()` names sound like getter, rename please to something like `calculateSalary()`